internal/resolver/dns/dns_resolver_test.go

```diff
@@ -736,12 +749,122 @@ func testDNSResolver(t *testing.T) {
+	newTimer = func(d time.Duration) *time.Timer {
+		// Will never fire on it's own, allows this test to call timer immediately.
+		t := time.NewTimer(time.Hour)
+		timerChan <- t
+		return t
+	}
+	tests := []struct {
+		target   string
+		addrWant []resolver.Address
+		scWant   string
+	}{
+		{
+			"foo.bar.com",
+			[]resolver.Address{{Addr: "1.2.3.4" + colonDefaultPort}, {Addr: "5.6.7.8" + colonDefaultPort}},
+			generateSC("foo.bar.com"),
+		},
+		{
+			"foo.bar.com:1234",
+			[]resolver.Address{{Addr: "1.2.3.4:1234"}, {Addr: "5.6.7.8:1234"}},
+			generateSC("foo.bar.com"),
+		},
+		{
+			"srv.ipv4.single.fake",
+			[]resolver.Address{{Addr: "2.4.6.8" + colonDefaultPort}},
+			generateSC("srv.ipv4.single.fake"),
+		},
+	}
+	for _, a := range tests {
+		b := NewBuilder()
+		cc := &testClientConn{target: a.target}
+		// Cause ClientConn to return an error.
+		cc.updateStateErr = balancer.ErrBadResolverState
+		r, err := b.Build(resolver.Target{Endpoint: a.target}, cc, resolver.BuildOptions{})
+		timer := <-timerChan
+		timer.Reset(0)
+		if err != nil {
+			t.Fatalf("%v\n", err)
+		}
+		var state resolver.State
+		var cnt int
+		for i := 0; i < 2000; i++ {
+			state, cnt = cc.getState()
+			if cnt > 0 {
+				break
+			}
+			time.Sleep(time.Millisecond)
+		}
+		if cnt == 0 {
+			t.Fatalf("UpdateState not called after 2s; aborting")
+		}
+		if !reflect.DeepEqual(a.addrWant, state.Addresses) {
+			t.Errorf("Resolved addresses of target: %q = %+v, want %+v", a.target, state.Addresses, a.addrWant)
+		}
+		sc := scFromState(state)
+		if a.scWant != sc {
+			t.Errorf("Resolved service config of target: %q = %+v, want %+v", a.target, sc, a.scWant)
+		}
+		// Cause timer to go off 10 times, and see if it calls updateState() correctly.
+		for i := 0; i < 10; i++ {
+			timer := <-timerChan
+			timer.Reset(0)
+			// Propagate.
+			time.Sleep(time.Millisecond)
+		}
+		cc.m1.Lock()
+		if cc.updateStateCalls != 12 {
+			cc.m1.Unlock()
+			t.Fatalf("Exponential backoff is not working as expected - should be updating state at least 10 total times instead of %d", cc.updateStateCalls)
+		}
+		cc.m1.Unlock()
+
+		// Update resolver.ClientConn to not return an error anymore - this should stop it from backing off.
+		cc.updateStateErr = nil
+		timer = <-timerChan
+		timer.Reset(0)
+		// Propagate the updated state.
+		time.Sleep(time.Millisecond * 3)
+
+		select {
+		case <-timerChan:
+			t.Error("Should not poll again after no more error")
+		default:
+		}
+		cc.m1.Lock()
+		if cc.updateStateCalls != 13 {
+			cc.m1.Unlock()
+			t.Fatalf("Exponential backoff is not working as expected - should stop backing off.")
+		}
+		cc.m1.Unlock()
+		r.Close()
+	}
+}
+
 func testDNSResolverWithSRV(t *testing.T) {
 	EnableSRVLookups = true
 	defer func() {
 		EnableSRVLookups = false
 	}()
 	defer leakcheck.Check(t)
+	nt := newTimer
+	defer func(func(d time.Duration) *time.Timer) {
+		newTimer = nt
+	}(nt)
+	newTimer = func(d time.Duration) *time.Timer {
```

Comment:
For functions that ignore their parameters, you can either not name them or use underscores:

```go
newTimer = func(time.Duration) *time.Timer {
```
or
```go
newTimer = func(_ time.Duration) *time.Timer {
```

This makes it clear that they are ignored, and that it is intentional.

Reply:
Changed to the second option of an underscore.